### DISCUSSION TITLE
LESSON 7: Foundry doesnt found tests

### DISCUSSION THREAD

**Discussion by nachoddiaz** [Cyfrin/foundry-full-course-cu#393] [Q&A] [answered]
Hello!

First of all, thanks Patrick and all the team for all the marvelous courses that you bring to us

- Command line and Error

When i run `forge test` ([https://www.youtube.com/watch?v=sas02qSFZ74&ab_channel=PatrickCollins](url) at this moment: 1:03:25), Foundry brings back "No tests found in project! Forge looks for functions that starts with `test`."

- My code

These are the relevant parts of:

FundMeTest.t.sol
```// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {Test, console} from "forge-std/Test.sol";
import {FundMe} from "../../src/FundMe.sol";
import {DeployFundMe} from "../../script/DeployFundMe.s.sol";

contract FundMeTest{
    FundMe fundMe;
    
    function setUp() external {
        DeployFundMe deployFundMe = new DeployFundMe();
        fundMe = deployFundMe.run();
    }
 
    function testMinFund() public {
        assertEq(fundMe.minUSD(), 50e18);
    }

```

DeployFundMe.s.sol

```
/ SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {Script} from "forge-std/Script.sol";
import {FundMe} from "../src/FundMe.sol";
import {HelperConfig} from "./HelperConfig.s.sol";

contract DeployFundMe is Script {
    function run() public returns (FundMe) {
        HelperConfig helperConfig = new HelperConfig();
        address PriceFeed = helperConfig.ActiveNetworkConfig();

        vm.startBroadcast();

        FundMe fundMe = new FundMe(PriceFeed);

        vm.stopBroadcast();
        return fundMe;
    }
}
```

- Steps I followed to find the solution

1. Ask ChatGPT -> It doesnt know about Foundry so it didn't give me a logical answer
2. General discussion tab and this unit tab -> there have been no problems related to this
3. Google/StackExchange -> All Q&A related to skip tests but none related to this

Hope somenone can help me, huge thanks.

**Comment by ChefAharoni** [comment]
The test contract should inherit from the test contract, like so:
`contract FundMeTest is Test { `

**Reply by nachoddiaz** [reply]
Sorry, i inherited the test contract but didnt write it here
```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {Test, console} from "forge-std/Test.sol";
import {FundMe} from "../../src/FundMe.sol";
import {DeployFundMe} from "../../script/DeployFundMe.s.sol";

contract FundMeTest is Test{
    FundMe fundMe;
    
    function setUp() external {
        DeployFundMe deployFundMe = new DeployFundMe();
        fundMe = deployFundMe.run();
    }
 
    function testMinFund() public {
        assertEq(fundMe.minUSD(), 50e18);
    }
```

it still shows me the following error _No tests found in project! Forge looks for functions that starts with test_ when I run **forge test**

**Comment by garyghayrat** [comment]
Hi, could you confirm that you're running the command in the right directory?
Also try `forge clean` `forge build` `forge test` in that order

**Reply by nachoddiaz** [reply]
Hi, yes, im running the command in the foundry-fundme directory which is where I have the directories named script, src, test... 

![Directory](https://github.com/Cyfrin/foundry-full-course-f23/assets/100040420/98184ada-b950-4911-bba5-d858f3fb1fa4)


Thanks for the proposal but I just tried it and it returns the same error
![Captura de pantalla 2023-07-20 235906](https://github.com/Cyfrin/foundry-full-course-f23/assets/100040420/9e4487b6-274f-4790-b451-48b46670094f)

**Reply by ChefAharoni** [reply]
Try changing the test's visibility to `external`. 

`public` should work as well, but sometimes its working funky.

**Reply by nachoddiaz** [reply]
I just change the visibility to external just in this function 
```
function testMinFund() external {
        assertEq(fundMe.minUSD(), 50e18);
    }
```

but it returns the same error.
Thx

**Reply by PatrickAlphaC** [reply]
Yep... We should figure this out. I've run into this before too. It's usually a capitalization error or something. Wondering what happens if you re-write these tests from scratch, what happens?

**Reply by ghost** [reply, marked as answer]
@nachoddiaz rename the test folder to "test" and try again.

**Reply by nachoddiaz** [reply]
Huge thanks, this is the answer. My test forlder name was "tests" and Foundry search for a folder named "test".

**Reply by albin-george-kurian** [reply]
rename the function name put test in prefix of function name 
```
function testFUNCTIONNAME() external{

//test cases

}
```

**Reply by Benji223** [reply]
worked

**Reply by chain-l** [reply]
Renaming test to test did not help but using "test" as a prefix to the setUp() function name helped.

**Reply by hoBabu1** [reply]
worked 
thanks a lot

**Comment by ChefAharoni** [comment]
Is it true just for this test? Have you tried adding more tests?

Probably not related, since there's not `vm.prank` in your test, but try also importing `StdCheats` and inheriting it in your contract. 
```solidity
import {StdCheats} from "forge-std/StdCheats.sol";

contract FundMeTest is Test, StdCheats {```

**Reply by nachoddiaz** [reply]
No, it happens with all tests, I just wrote one to simplify the problem.
Also i tried with StdCheats and it remains the same
Thanks

**Reply by prakhar-kt** [reply]
My test folder was named "test". Yet I renamed it to "test" and it worked!!

**Comment by KarthikeyaGundumogula** [comment]
I tried renaming "test" folder to "test" it doesn't worked for me, after adding "test" prefix to every test function name it worked fine.

**Comment by Vodvalex** [comment]
Thank you @chain-l  for your post. I had the same problem and changing to the  testSetUp() helped very well.

**Comment by Lukelr12** [comment]
I have tried everything on this list and still no luck, getting the same error 

No tests found in project! Forge looks for functions that starts with `test`.
Error: 
No tests to run

I am also using WSL if anyone would be willing to help me out here that would be apprecated. 

my code 

''' 
// SPDX-License-Identifier: MIT 

pragma solidity ^0.8.18;

import {Test} from "forge-std/Test.sol";

contract FundMeTest is Test{
    uint256 number = 1;

    function setUp() external {
        number = 2;
    }

    function testDemo() public {
        assertEq(number, 2);
    }
}
'''

**Reply by EngrPips** [reply]
Show your text editor as I show mine below 

<img width="1440" alt="Screenshot 2024-09-30 at 10 39 05" src="https://github.com/user-attachments/assets/b1a5c6db-4cb7-4f62-9951-15a7f62a030b">

**Reply by Lukelr12** [reply]
![Screenshot 2024-09-30 024251](https://github.com/user-attachments/assets/31ae36df-f342-4d0c-855f-8af03f123a84)

**Reply by EngrPips** [reply]
Everything seems fine. Please delete your cache folder, then run `foundryup`, then run `forge clean`, then retry the test. Let's see how that will go.

**Reply by Lukelr12** [reply]
same out come No tests found in project! Forge looks for functions that starts with `test`.
Error: 
No tests to run

**Reply by EngrPips** [reply]
Run `dir` in your terminal and let's see what will be the response.

**Reply by Lukelr12** [reply]
the responce is: README.md  foundry.toml  lib  script  src  test

**Reply by EngrPips** [reply]
I am unsure what you got wrong here but this seems like a `wsl` issue and not a foundry. you initialized your repo with `forge init`, right?

**Reply by Lukelr12** [reply]
yep, I had to use the --force. now that reintialied forge init --force again it worked

**Reply by EngrPips** [reply]
Interesting, I am glad you figured it out.

**Reply by Lukelr12** [reply]
![Screenshot 2024-09-30 033005](https://github.com/user-attachments/assets/7143cfe4-4d6f-4982-8a6b-8d14c9b508e4)

**Reply by EngrPips** [reply]
Keep crushing it.

**Reply by Lukelr12** [reply]
i really appreciate you're help!

**Reply by EngrPips** [reply]
Anytime, Friend.
